projects/storefrontlib/cms-components/misc/icon/icon.model.ts

```diff
@@ -43,6 +43,8 @@ export enum ICON_TYPE {
   EYE_SLASH = 'EYE_SLASH',
 }
 
+export type ICON_TYPE_STRING = ICON_TYPE | string;
```

Comment:
Added this, because both `IconLoaderService` and `IconComponent` where already using something similar. Either inline or as declared type.

Not sure about the name. Maybe `ICON_TYPE_OR_STRING` or `ICON_TYPISH` instead?

Reply:
Replacing existing types with this new type creates a series of breaking changes.  We should revert this change if we are aiming to release in a minor version.

Reply:
I don't think so. I've only used `ICON_TYPE_STRING` in locations that where already using `ICON_TYPE | string`, either directly or via a type alias. TS does not care how the type is declared, it only cares that it's `ICON_TYPE | string`.

In icon.component.ts it's a little tricky to see, because the old code redefined `ICON_TYPE`:
```
import { ICON_TYPE as DEFAULT_ICON_TYPE } from './icon.model';
type ICON_TYPE = DEFAULT_ICON_TYPE | string;
```

So in that class, the old ICON_TYPE is the same as the new `ICON_TYPE_STRING`.

I thought redefining the meaning of a type would be very confusing. Or is this a common practice?